src/core/InformationPanel/InformationPanel.tsx

```diff
@@ -0,0 +1,166 @@
+   * @default false
+   */
+  isOpen?: boolean;
+  /**
+   * Orientation of the panel.
+   * @default 'vertical'
+   */
+  orientation?: 'vertical' | 'horizontal';
+  /**
+   * Callback fired when close icon is clicked.
+   */
+  onClose?: () => void;
+  /**
+   * Custom header action buttons shown alongside the close button.
+   * @example
+   * [
+   *   <IconButton styleType='borderless' onClick={() => {}}><SvgEdit /></IconButton>,
+   *   <IconButton styleType='borderless' onClick={() => {}}><SvgDelete /></IconButton>,
+   * ]
+   */
+  headerActions?: React.ReactNode[];
+  /**
+   * Set to false to make the panel non-resizable.
+   * @default true
+   */
+  resizable?: boolean;
+  /**
+   * Content of the panel.
+   */
+  children?: React.ReactNode;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * InformationPanel provides a way to view additional content or attributes
+ * without leaving the main content area (e.g. Table).
+ *
+ * The panel appears as an overlay on top of the main content.
+ * @example
+ * <InformationPanelWrapper>
+ *   <Table ... /> // component to overlay
+ *   <InformationPanel
+ *     label={<Text variant='subheading'>InfoPanel heading</Text>}
+ *     onClose={() => {}}
+ *     isOpen={true}
+ *   >
+ *     <span>Info panel content</span>
+ *   </InformationPanel>
+ * </InformationPanelWrapper>
+ */
+export const InformationPanel = React.forwardRef<
+  HTMLDivElement,
+  InformationPanelProps
+>((props, ref) => {
+  const {
+    className,
+    label,
+    isOpen = false,
+    onClose,
+    orientation = 'vertical',
+    headerActions,
+    resizable = true,
+    children,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  const infoPanelRef = React.useRef<HTMLDivElement>(null);
+  const refs = useMergedRefs(ref, infoPanelRef);
+
+  const onResize = React.useCallback(
+    (e: PointerEvent) => {
+      e.preventDefault();
+      if (!infoPanelRef.current) {
+        return;
+      }
+      const { right, bottom } = infoPanelRef.current.getBoundingClientRect();
+      if (orientation === 'vertical') {
+        infoPanelRef.current.style.width = `${right - e.clientX}px`;
+      } else {
+        infoPanelRef.current.style.height = `${bottom - e.clientY}px`;
+      }
+    },
+    [orientation],
+  );
+
+  return (
+    <div
+      className={cx(
+        'iui-information-panel',
+        {
+          'iui-right': orientation === 'vertical',
+          'iui-bottom': orientation === 'horizontal',
+          'iui-visible': isOpen,
+        },
+        className,
+      )}
+      ref={refs}
+      {...rest}
+    >
+      {resizable && (
+        <div
+          className='iui-resizer'
+          onPointerDown={(e) => {
+            if (!infoPanelRef.current) {
+              return;
+            }
+            if (e.button != undefined && e.button !== 0) {
+              return;
+            }
+            e.preventDefault();
+            e.stopPropagation();
+            infoPanelRef.current.ownerDocument.addEventListener(
```

Comment:
Can't you use our hook for this?

Reply:
I know, you would have it outside. But now started to think whether you need `addEventListener`, can't you have like a flag `isResizing` and have react props `onPointerDown` and `onPointerMove`?

Reply:
`onPointerDown` and `onPointerMove` would need to be applied on some element. In this case, I'm adding the listeners to document, so it wouldn't work.

Fwiw, I added `{ once: true }` so there is no concern of the listener not getting removed.